packages/effect/src/internal/random.ts

```diff
@@ -79,6 +81,12 @@ const shuffleWith = <A>(
   )
 }
 
+export const choice = <A>(elements: array.NonEmptyReadonlyArray<A>): Effect.Effect<A> =>
```

Comment:
Will need to move this to the "defaultServices" module.

Reply:
Should the elements array really be `NonEmpty`?

Reply:
Yes, otherwise you will need to return an Option or fail with a NoSuchElementException.

Reply:
Fixed

Reply:
wondering if we shouldn't make this conditional, like `X extends ReadonlyArray` and if not empty then return `Effect<A>`, otherwise `Effect<A, NoSuchElement...>`, forcing `NonEmpty` proved tedious to use in other APIs

Reply:
Updated

Reply:
What about widening `Array` to `Iterable`?

Reply:
The length must be known upfront so I don't it's possible to make `choice` work for any iterable.

Reply:
You can materialize an `Iterable` to an `Array` before accepting an element. 
I think this request will appear in the future anyway.

Reply:
Suggest an edit then lol
also I don't think this api is going to be very popular, like not even top 100, it's quite niche in the end of the day

Reply:
My problem with allowing any iterable is that we'd allow infinite iterators as well and in such a case it would throw an exception (by trying it locally, the `Array.from` gives up after ~125M iterations and throws a `RangeError` exception). If we're okay with this, I'll add the support, document this edge-case behaviour in the doc-string and convert the exception to `Effect.die`. WDYT?

Reply:
> My problem with allowing any iterable is that we'd allow infinite iterators as well and in such a case it would throw an exception (by trying it locally, the `Array.from` gives up after ~125M iterations and throws a `RangeError` exception). If we're okay with this, I'll add the support, document this edge-case behaviour in the doc-string and convert the exception to `Effect.die`. WDYT?

kind of the same for every single api that allows for an iterable and we have plenty, so I'd not consider this an issue. You can have `X extends Iterable<any>` and then use a conditional to branch the cases

Reply:
ok